### ISSUE TITLE
Missing workflow initialisation

### ISSUE THREAD

**Issue by MarcinOkon** [sergeyshushlyapin/Sitecore.FakeDb#118]
Hi
I have created a template with a default workflow (defaultworkflow field set)
I would then expect that a new item based on that template would not only have the defaultworkflow but also the workflow field set.

```c#
[Test]
public void WorkflowTest()
{
	var templateId = ID.NewID;

	using (var db = new Db
	{
	  new DbTemplate(templateId)
	  {
		  {FieldIDs.DefaultWorkflow, "{E02A54E4-1037-4569-A735-F582B8ABA8A4}"}
	  },
	  new DbItem("home", ID.NewID,templateId)
	})
	{
		Item home = db.GetItem("/sitecore/content/home");

		Assert.AreEqual(templateId, home.TemplateID);
		Assert.AreEqual("{E02A54E4-1037-4569-A735-F582B8ABA8A4}", home[FieldIDs.DefaultWorkflow]);
		Assert.AreEqual("{E02A54E4-1037-4569-A735-F582B8ABA8A4}", home[FieldIDs.Workflow]); //-> fails here
	}
}
```

Ideally I think it would be good to have general support for workflows like having a sampleworkflow support with an initial workflowstate and the possibility to check if a certain state is final and so on

**Comment by pveller** [comment]
We sure can add it. One tricky part will be workflow comments templates in 8+ that also affected workflow APIs.

**Comment by sergeyshushlyapin** [comment]
I tried to reproduce the scenario on real Sitecore instances (8.0 and 8.1). The *Workflow* section is hidden on the template level so there is no possibility to set the `DefaultWorkflow` field from the UI at all...       I'm not sure if that is a bug or feature; I'll clarify this.

When I try to set the `DefaultWorkflow` on the template programmatically and then create an item based on that template, all the workflow fields in the result item are empty. It behaves exactly as in the example you posted which makes me think we should do nothing with FakeDb here.

As far as I can see, you're trying to create a template in the code that would correspond to the real item template. It may lead to the situation when the tests might be grow too much. Would't it be enough *for the test case* just to have the workflow configured *somehow* (e.g. directly on the item level), so that you don't need to create the template manually at all:
```csharp
[Fact]
public void WorkflowTest()
{
  using (var db = new Db
    {
      new DbItem("home") { { FieldIDs.Workflow, "{E02A54E4-1037-4569-A735-F582B8ABA8A4}" } }
    })
  {
    Item home = db.GetItem("/sitecore/content/home");

    Assert.Equal("{E02A54E4-1037-4569-A735-F582B8ABA8A4}", home[FieldIDs.Workflow]);
  }
}
```

**Comment by pveller** [comment]
Workflows should be set on standard values I believe, not template items. Am I missing something?

**Comment by sergeyshushlyapin** [comment]
Nope, it seems that's me who's missing something ;). Thanks @pveller, I'll take a look once again.

**Comment by MarcinOkon** [comment]
> Would't it be enough for the test case just to have the workflow configured somehow (e.g. directly on the item level), so that you don't need to create the template manually at all:

The reason I want to set the default workflow on the template instead of setting the workflow directly on the item is that I actually want to check a clone of an item which is created within the code.

Thats why it would be really helpful if the new item would also have its initial workflow state set (based on some sample workflow). Although I guess this would involve a lot of work to implement properly.

**Comment by sergeyshushlyapin** [comment]
@MarcinOkon, thanks for clarifying this, it definitely may sense. I need to investigate the default Sitecore API behaviour and then come up with the solution.

**Comment by sergeyshushlyapin** [comment]
The minimal implementation is available in [v0.37.0](https://www.nuget.org/packages/Sitecore.FakeDb/).
